doc/docs/Python_Tutorials/Eigenmode_Source.md

```diff
@@ -229,62 +229,97 @@ Increasing the size of the cell improves results at the expense of a larger simu
-                              eig_kpoint=k_point,
-                              eig_band=1,
-                              eig_parity=mp.EVEN_Y+mp.ODD_Z if rot_angle == 0 else mp.ODD_Z,
-                              eig_match_freq=True)]
-
-sim = mp.Simulation(cell_size=cell_size,
-                    resolution=resolution,
-                    boundary_layers=pml_layers,
-                    sources=sources,
-                    k_point=k_point,
-                    default_material=default_material,
-                    symmetries=[mp.Mirror(mp.Y)] if rot_angle == 0 else [])
+n_mat = 1.5  # refractive index of homogeneous material
+default_material = mp.Medium(index=n_mat)
 
-sim.run(until=100)
+k_point = mp.Vector3(n_mat * frequency, 0, 0).rotate(
+    mp.Vector3(0, 0, 1),
+    incident_angle
+)
 
-nonpml_vol = mp.Volume(center=mp.Vector3(), size=mp.Vector3(10,10,0))
+if incident_angle == 0:
+    direction = mp.AUTOMATIC
+    eig_parity = mp.EVEN_Y + mp.ODD_Z
+    symmetries = [mp.Mirror(mp.Y)]
+    eig_vol = None
+else:
+    direction = mp.NO_DIRECTION
+    eig_parity = mp.ODD_Z
+    symmetries = []
+    eig_vol = mp.Volume(
+        center=mp.Vector3(),
+        size=mp.Vector3(0, 1 / resolution_um, 0)
+    )
+
+sources = [
+    mp.EigenModeSource(
+        src=mp.ContinuousSource(frequency),
+        center=mp.Vector3(),
+        size=mp.Vector3(0, size_um, 0),
+        direction=direction,
+        eig_kpoint=k_point,
+        eig_band=1,
+        eig_parity=eig_parity,
+        eig_vol=eig_vol
+    )
+]
+
+sim = mp.Simulation(
+    cell_size=cell_size,
+    resolution=resolution_um,
+    boundary_layers=pml_layers,
+    sources=sources,
+    k_point=k_point,
+    default_material=default_material,
+    symmetries=symmetries
+)
+
+sim.run(until=23.56)
+
+output_plane = mp.Volume(
+    center=mp.Vector3(),
+    size=mp.Vector3(size_um, size_um, 0)
+)
+
+fig, ax = plt.subplots()
+sim.plot2D(fields=mp.Ez, output_plane=output_plane, ax=ax)
+fig.savefig("planewave_source.png", bbox_inches="tight")
+```
 
-sim.plot2D(fields=mp.Ez,
-           output_plane=nonpml_vol)
+There are several items to note in this script.
 
-if mp.am_master():
-    plt.axis('off')
-    plt.savefig('pw.png',bbox_inches='tight')
-```
+1. The line source spans the *entire* length of the cell. Planewave sources extending into the PML region must include `is_integrated=True` in the source definition.
+2. The wavevector of the planewave $\vec{k}$ is defined using the `k_point` which specifies Bloch-periodic boundaries. PML boundaries are used only along the $x$ direction.
+3. The `eig_vol` parameter of the `EigenModeSource` defines a volume that is one-pixel thick in the direction of the line monitor. This is necessary to ensure that the planewave's wavevector `eig_kpoint` which is defined in the continuum rather than for a system with discrete translational symmetry is correctly interpreted by MPB.
```

Comment:
```suggestion
3. The `eig_vol` parameter of the `EigenModeSource` defines a volume that is one-pixel thick in the direction of the line monitor. This is necessary to ensure that the`eig_kpoint` is interpreted by MPB as the wavevector of a planewave — that is, so it is the wavevector as defined for a homogeneous material (continuous translational symmetry) rather than as a Bloch wavevector for a system with discrete translational symmetry.
```